core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -2033,7 +2033,9 @@ wasm_lookup_table(const WASMModuleInstance *module_inst, const char *name)
 static bool
 clear_wasi_proc_exit_exception(WASMModuleInstance *module_inst)
 {
-#if WASM_ENABLE_LIBC_WASI != 0
+    /* With WASI threads, proc_exit exception is not cleared because it has to
```

Comment:
should we then capture the exception and clear it for each module instances?

Reply:
I think I need to modify `wasm_runtime_call_wasm()` to return `true` in case the exception encountered is `proc_exit`

Reply:
my expectation on `proc_exit` is:

* wasm_application_execute_main etc succeeds without an exception
* exit code is propagated here https://github.com/bytecodealliance/wasm-micro-runtime/blob/219713bfbc5196ee7c91fe78d18ddceed069c179/product-mini/platforms/posix/main.c#L669

Reply:
> I get an exception so the runtime doesn't return 0 at the end 

> it seems like the exit code gets already propagated properly

I'm a bit confused, so does it return the exit code being passed to `proc_exit()`?

Reply:
yes, the return code is set when `proc_exit` is called https://github.com/bytecodealliance/wasm-micro-runtime/blob/903f5212f194c2709b32631bd32936fdcfed27af/core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c#L989 and then retrieved here https://github.com/bytecodealliance/wasm-micro-runtime/blob/219713bfbc5196ee7c91fe78d18ddceed069c179/product-mini/platforms/posix/main.c#L669
I tested it by calling `__wasi_proc_exit(exit_code)` with different exit codes and I see the `iwasm` executable returning with the proper exit codes